### PR TITLE
[Fabric] Introduce clearTextOnFocus property for TextInput

### PR DESCRIPTION
**Pull request by danielayala94**
### Type of Change

New feature.

### Why

This is part of our Parity efforts, in this case to enable textInput to clear text when it gains focus.

### What

Added a member variable to track if clearTextOnFocus is true, included the respective check in updateProps and clear the text field when it gains focus AND the prop is set to true.

## Screenshots

https://github.com/microsoft/react-native-windows/assets/14967941/03c8edd2-3c35-467d-bc05-9c94a30a31e0

## Testing
- Added a TextInput into samples/textInput.tsx demonstrating the new prop. Ran playground-composition to test the feature and record the screenshot video.
- Introduced two tests in the E2E Fabric Test App to verify behavior when clearTextOnFocus is true and false respectively.

## Changelog
No?
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13227)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.h [hunk 1]
```diff
@@ -131,6 +131,7 @@ struct WindowsTextInputComponentView
   bool m_drawing{false};
   bool m_clearTextOnSubmit{false};
   bool m_multiline{false};
+  bool m_clearTextOnFocus{false};
```
Comment (acoates-ms) [inline vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.h:134]: You shouldn't need to duplicate the props.  You can already access this using `windowsTextInputProps().clearTextOnFocus`

File: packages/playground/Samples/textinput.tsx [hunk 2]
```diff
@@ -120,6 +120,11 @@ export default class Bootstrap extends React.Component<{}, any> {
             maxLength={10}
             keyboardType="numeric"
           />
+          <TextInput
```
Comment (acoates-ms) [inline packages/playground/Samples/textinput.tsx:123]: You should add this as an example to TextInputExample.windows.js - And add a new test to the E2E test app to verify that it works.

**Comment by jonthysell** [comment]
You'll need to run `yarn change` at the root of your repo to create a change-file for the PR checks.

**Comment by jonthysell** [comment]
@danielayala94 Also, int the future, you should create the branch on your own fork of the microsoft repo.

**Comment by TatianaKapos** [comment]
> @danielayala94 Also, int the future, you should create the branch on your own fork of the microsoft repo.

Instructions for Contributing here: https://github.com/microsoft/react-native-windows/blob/main/docs/contributing.md :)

File: vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp [hunk 3]
```diff
@@ -945,6 +945,10 @@ void WindowsTextInputComponentView::onFocusGained() noexcept {
     LRESULT lresult;
     DrawBlock db(*this);
     m_textServices->TxSendMessage(WM_SETFOCUS, 0, 0, &lresult);
+
+    if (m_props->clearTextOnFocus) {
```
Comment (danielayala94) [inline vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp:949]: Right now, this is a build error. Not sure how I didn't catch this before.

Found the right way to go, will update in next iteration.

**Comment by TatianaKapos** [comment]
To get the Hermes E2ETests to run, you'll have to update the snapshot! Instructions here:https://github.com/microsoft/react-native-windows/wiki/E2E-Testing-(Fabric)#writing-snapshot-tests, I usually update them by grabbing the snapshot files from the failed pipeline

**Review by marlenecota — APPROVED** [review]

**Review by jonthysell — APPROVED** [review]
LGTM

**Pull request merged**